middle_end/flambda/from_lambda/closure_conversion.ml

```diff
@@ -47,6 +47,19 @@ let symbol_for_ident acc env id =
   let symbol = Env.symbol_for_global' env id in
   use_of_symbol_as_simple acc symbol
 
+let register_set_of_closures_as_symbols acc set_of_closures
+      : Acc.t * Symbol.t Closure_id.Lmap.t =
+  let symbols =
+    Set_of_closures.function_decls set_of_closures
+    |> Function_declarations.funs_in_order
+    |> Closure_id.Lmap.mapi (fun closure_id _ ->
+        Symbol.create (Compilation_unit.get_current_exn ())
+          (Linkage_name.create
+             (Variable.unique_name (Closure_id.unwrap closure_id))))
```

Comment:
You should probably use the same code as `Simplify_set_of_closures.simplify_and_lift_set_of_closures`, which would be `Linkage_name.create (Closure_id.to_string closure_id)` (I think you can skip the extra `Closure_id.rename` here).